// @injectivelabs/sdk-ts/dist/core/modules/authz/msgs/MsgRevoke.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/cosmos/authz/v1beta1/tx_pb");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
const MsgBase_1 = require("../../MsgBase");
/**
 * @category Messages
 */
class MsgRevoke extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgRevoke(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgRevoke();
        message.setGrantee(params.grantee);
        message.setGranter(params.granter);
        message.setMsgTypeUrl(params.messageType);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmos.authz.v1beta1.MsgRevoke' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'cosmos-sdk/MsgRevoke',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmos.authz.v1beta1.MsgRevoke' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmos.authz.v1beta1.MsgRevoke',
            message: proto,
        };
    }
}
exports.default = MsgRevoke;
