// @injectivelabs/sdk-ts/dist/client/chain/grpc/ChainGrpcBankApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcBankApi = void 0;
const query_pb_service_1 = require("@injectivelabs/chain-api/cosmos/bank/v1beta1/query_pb_service");
const query_pb_1 = require("@injectivelabs/chain-api/cosmos/bank/v1beta1/query_pb");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const pagination_1 = require("../../../utils/pagination");
const transformers_1 = require("../transformers");
const exceptions_1 = require("@injectivelabs/exceptions");
const types_1 = require("../types");
/**
 * @category Chain Grpc API
 */
class ChainGrpcBankApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.ChainModule.Bank;
    }
    fetchModuleParams() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryParamsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Params);
                return transformers_1.ChainGrpcBankTransformer.moduleParamsResponseToModuleParams(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchBalance({ accountAddress, denom, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryBalanceRequest();
            request.setAddress(accountAddress);
            request.setDenom(denom);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Balance);
                return transformers_1.ChainGrpcBankTransformer.balanceResponseToBalance(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchBalances(address) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryAllBalancesRequest();
            request.setAddress(address);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.AllBalances);
                return transformers_1.ChainGrpcBankTransformer.balancesResponseToBalances(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchTotalSupply(pagination) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryTotalSupplyRequest();
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.TotalSupply);
                return transformers_1.ChainGrpcBankTransformer.totalSupplyResponseToTotalSupply(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.ChainGrpcBankApi = ChainGrpcBankApi;
