// @injectivelabs/sdk-ts/dist/client/chain/grpc/ChainGrpcWasmApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcWasmApi = void 0;
const query_pb_service_1 = require("@injectivelabs/chain-api/cosmwasm/wasm/v1/query_pb_service");
const query_pb_1 = require("@injectivelabs/chain-api/cosmwasm/wasm/v1/query_pb");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const transformers_1 = require("../transformers");
const pagination_1 = require("../../../utils/pagination");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Chain Grpc API
 */
class ChainGrpcWasmApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.ChainModule.Wasm;
    }
    fetchContractAccountsBalance({ contractAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryAllContractStateRequest();
            request.setAddress(contractAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.AllContractState);
                return transformers_1.ChainGrpcWasmTransformer.allContractStateResponseToContractAccountsBalanceWithPagination(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchContractInfo(contractAddress) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryAllContractStateRequest();
            request.setAddress(contractAddress);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ContractInfo);
                const contractInfo = response.getContractInfo();
                if (!contractInfo) {
                    return;
                }
                return transformers_1.ChainGrpcWasmTransformer.contactInfoResponseToContractInfo(contractInfo);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchContractHistory(contractAddress) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryContractHistoryRequest();
            request.setAddress(contractAddress);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ContractHistory);
                return transformers_1.ChainGrpcWasmTransformer.contactHistoryResponseToContractHistory(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchSmartContractState(contractAddress, query) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QuerySmartContractStateRequest();
            request.setAddress(contractAddress);
            if (query) {
                request.setQueryData(query);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.SmartContractState);
                return response.toObject();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchRawContractState(contractAddress, query) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryRawContractStateRequest();
            request.setAddress(contractAddress);
            if (query) {
                request.setQueryData(query);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.RawContractState);
                return response.toObject();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchContractCodes(pagination) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryCodesRequest();
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Codes);
                return transformers_1.ChainGrpcWasmTransformer.contractCodesResponseToContractCodes(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchContractCode(codeId) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryCodeRequest();
            request.setCodeId(codeId);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Code);
                return transformers_1.ChainGrpcWasmTransformer.contractCodeResponseToContractCode(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchContractCodeContracts(codeId, pagination) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryContractsByCodeRequest();
            request.setCodeId(codeId);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ContractsByCode);
                return transformers_1.ChainGrpcWasmTransformer.contractByCodeResponseToContractByCode(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.ChainGrpcWasmApi = ChainGrpcWasmApi;
