// @injectivelabs/sdk-ts/dist/client/chain/grpc/ChainGrpcDistributionApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcDistributionApi = void 0;
const query_pb_1 = require("@injectivelabs/chain-api/cosmos/distribution/v1beta1/query_pb");
const query_pb_service_1 = require("@injectivelabs/chain-api/cosmos/distribution/v1beta1/query_pb_service");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const transformers_1 = require("../transformers");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Chain Grpc API
 */
class ChainGrpcDistributionApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.ChainModule.Distribution;
    }
    fetchModuleParams() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryParamsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Params);
                return transformers_1.ChainGrpcDistributionTransformer.moduleParamsResponseToModuleParams(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchDelegatorRewardsForValidator({ delegatorAddress, validatorAddress, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegationRewardsRequest();
            request.setValidatorAddress(validatorAddress);
            request.setDelegatorAddress(delegatorAddress);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.DelegationRewards);
                return transformers_1.ChainGrpcDistributionTransformer.delegationRewardResponseToReward(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e);
            }
        });
    }
    fetchDelegatorRewardsForValidatorNoThrow({ delegatorAddress, validatorAddress, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegationRewardsRequest();
            request.setValidatorAddress(validatorAddress);
            request.setDelegatorAddress(delegatorAddress);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.DelegationRewards);
                return transformers_1.ChainGrpcDistributionTransformer.delegationRewardResponseToReward(response);
            }
            catch (e) {
                if (e.message.includes('does not exist')) {
                    return [];
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e);
            }
        });
    }
    fetchDelegatorRewards(injectiveAddress) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegationTotalRewardsRequest();
            request.setDelegatorAddress(injectiveAddress);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.DelegationTotalRewards);
                return transformers_1.ChainGrpcDistributionTransformer.totalDelegationRewardResponseToTotalReward(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e);
            }
        });
    }
    fetchDelegatorRewardsNoThrow(injectiveAddress) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegationTotalRewardsRequest();
            request.setDelegatorAddress(injectiveAddress);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.DelegationTotalRewards);
                return transformers_1.ChainGrpcDistributionTransformer.totalDelegationRewardResponseToTotalReward(response);
            }
            catch (e) {
                if (e.message.includes('does not exist')) {
                    return [];
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e);
            }
        });
    }
}
exports.ChainGrpcDistributionApi = ChainGrpcDistributionApi;
