// @injectivelabs/sdk-ts/dist/client/chain/grpc/ChainGrpcInsuranceFundApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcInsuranceFundApi = void 0;
const query_pb_service_1 = require("@injectivelabs/chain-api/injective/insurance/v1beta1/query_pb_service");
const query_pb_1 = require("@injectivelabs/chain-api/injective/insurance/v1beta1/query_pb");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const ChainGrpcInsuranceFundTransformer_1 = require("../transformers/ChainGrpcInsuranceFundTransformer");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Chain Grpc API
 */
class ChainGrpcInsuranceFundApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.ChainModule.InsuranceFund;
    }
    fetchModuleParams() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryInsuranceParamsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.InsuranceParams);
                return ChainGrpcInsuranceFundTransformer_1.ChainGrpcInsuranceFundTransformer.moduleParamsResponseToModuleParams(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchInsuranceFunds() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryInsuranceFundsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.InsuranceFunds);
                return ChainGrpcInsuranceFundTransformer_1.ChainGrpcInsuranceFundTransformer.insuranceFundsResponseToInsuranceFunds(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchInsuranceFund(marketId) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryInsuranceFundRequest();
            request.setMarketId(marketId);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.InsuranceFund);
                return ChainGrpcInsuranceFundTransformer_1.ChainGrpcInsuranceFundTransformer.insuranceFundResponseToInsuranceFund(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchEstimatedRedemptions({ marketId, address, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryEstimatedRedemptionsRequest();
            request.setMarketid(marketId);
            request.setAddress(address);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.EstimatedRedemptions);
                return ChainGrpcInsuranceFundTransformer_1.ChainGrpcInsuranceFundTransformer.estimatedRedemptionsResponseToEstimatedRedemptions(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchPendingRedemptions({ marketId, address, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryPendingRedemptionsRequest();
            request.setMarketid(marketId);
            request.setAddress(address);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.PendingRedemptions);
                return ChainGrpcInsuranceFundTransformer_1.ChainGrpcInsuranceFundTransformer.redemptionsResponseToRedemptions(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.ChainGrpcInsuranceFundApi = ChainGrpcInsuranceFundApi;
