// @injectivelabs/sdk-ts/dist/client/chain/grpc/ChainGrpcExchangeApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcExchangeApi = void 0;
const query_pb_service_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/query_pb_service");
const query_pb_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/query_pb");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const transformers_1 = require("../transformers");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Chain Grpc API
 */
class ChainGrpcExchangeApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.ChainModule.Exchange;
    }
    fetchModuleParams() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryExchangeParamsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.QueryExchangeParams);
                return transformers_1.ChainGrpcExchangeTransformer.moduleParamsResponseToParams(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
    fetchModuleState() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryModuleStateRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ExchangeModuleState);
                return response.getState().toObject(); /* TODO */
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
    fetchFeeDiscountSchedule() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryFeeDiscountScheduleRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.FeeDiscountSchedule);
                return transformers_1.ChainGrpcExchangeTransformer.feeDiscountScheduleResponseToFeeDiscountSchedule(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
    fetchFeeDiscountAccountInfo(injectiveAddress) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryFeeDiscountAccountInfoRequest();
            request.setAccount(injectiveAddress);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.FeeDiscountAccountInfo);
                return transformers_1.ChainGrpcExchangeTransformer.feeDiscountAccountInfoResponseToFeeDiscountAccountInfo(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
    fetchTradingRewardsCampaign() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryTradeRewardCampaignRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.TradeRewardCampaign);
                return transformers_1.ChainGrpcExchangeTransformer.tradingRewardsCampaignResponseToTradingRewardsCampaign(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
    fetchTradeRewardPoints(injectiveAddresses) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryTradeRewardPointsRequest();
            request.setAccountsList(injectiveAddresses);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.TradeRewardPoints);
                return response.getAccountTradeRewardPointsList();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
    fetchPendingTradeRewardPoints(injectiveAddresses, timestamp) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryTradeRewardPointsRequest();
            request.setAccountsList(injectiveAddresses);
            if (timestamp) {
                request.setPendingPoolTimestamp(timestamp);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.PendingTradeRewardPoints);
                return response.getAccountTradeRewardPointsList();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
    fetchPositions() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryPositionsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Positions);
                return transformers_1.ChainGrpcExchangeTransformer.positionsResponseToPositions(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
    fetchSubaccountTradeNonce(subaccountId) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QuerySubaccountTradeNonceRequest();
            request.setSubaccountId(subaccountId);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.SubaccountTradeNonce);
                return response.toObject();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: types_1.ChainModule.Exchange,
                });
            }
        });
    }
}
exports.ChainGrpcExchangeApi = ChainGrpcExchangeApi;
