// @injectivelabs/sdk-ts/dist/client/chain/grpc/ChainGrpcStakingApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcStakingApi = void 0;
const query_pb_1 = require("@injectivelabs/chain-api/cosmos/staking/v1beta1/query_pb");
const query_pb_service_1 = require("@injectivelabs/chain-api/cosmos/staking/v1beta1/query_pb_service");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const pagination_1 = require("../../../utils/pagination");
const transformers_1 = require("../transformers");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Chain Grpc API
 */
class ChainGrpcStakingApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.ChainModule.Staking;
    }
    fetchModuleParams() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryParamsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Params);
                return transformers_1.ChainGrpcStakingTransformer.moduleParamsResponseToModuleParams(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchPool() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryPoolRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Pool);
                return transformers_1.ChainGrpcStakingTransformer.poolResponseToPool(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchValidators(pagination) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryValidatorsRequest();
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Validators);
                return transformers_1.ChainGrpcStakingTransformer.validatorsResponseToValidators(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchValidator(address) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryValidatorRequest();
            request.setValidatorAddr(address);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Validator);
                return transformers_1.ChainGrpcStakingTransformer.validatorResponseToValidator(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchValidatorDelegations({ validatorAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryValidatorDelegationsRequest();
            request.setValidatorAddr(validatorAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ValidatorDelegations);
                return transformers_1.ChainGrpcStakingTransformer.delegationsResponseToDelegations(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchValidatorDelegationsNoThrow({ validatorAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryValidatorDelegationsRequest();
            request.setValidatorAddr(validatorAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ValidatorDelegations);
                return transformers_1.ChainGrpcStakingTransformer.delegationsResponseToDelegations(response);
            }
            catch (e) {
                if (e.message.includes('does not exist')) {
                    return { delegations: [], pagination: { total: 0, next: '' } };
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchValidatorUnbondingDelegations({ validatorAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryValidatorUnbondingDelegationsRequest();
            request.setValidatorAddr(validatorAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ValidatorUnbondingDelegations);
                return transformers_1.ChainGrpcStakingTransformer.unBondingDelegationsResponseToUnBondingDelegations(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchValidatorUnbondingDelegationsNoThrow({ validatorAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryValidatorUnbondingDelegationsRequest();
            request.setValidatorAddr(validatorAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ValidatorUnbondingDelegations);
                return transformers_1.ChainGrpcStakingTransformer.unBondingDelegationsResponseToUnBondingDelegations(response);
            }
            catch (e) {
                if (e.message.includes('does not exist')) {
                    return { unbondingDelegations: [], pagination: { total: 0, next: '' } };
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchDelegation({ injectiveAddress, validatorAddress, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegationRequest();
            request.setDelegatorAddr(injectiveAddress);
            request.setValidatorAddr(validatorAddress);
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Delegation);
                return transformers_1.ChainGrpcStakingTransformer.delegationResponseToDelegation(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchDelegations({ injectiveAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegatorDelegationsRequest();
            request.setDelegatorAddr(injectiveAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.DelegatorDelegations);
                return transformers_1.ChainGrpcStakingTransformer.delegationsResponseToDelegations(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchDelegationsNoThrow({ injectiveAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegatorDelegationsRequest();
            request.setDelegatorAddr(injectiveAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.DelegatorDelegations);
                return transformers_1.ChainGrpcStakingTransformer.delegationsResponseToDelegations(response);
            }
            catch (e) {
                if (e.message.includes('does not exist')) {
                    return { delegations: [], pagination: { total: 0, next: '' } };
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchDelegators({ validatorAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryValidatorDelegationsRequest();
            request.setValidatorAddr(validatorAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ValidatorDelegations);
                return transformers_1.ChainGrpcStakingTransformer.delegationsResponseToDelegations(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchDelegatorsNoThrow({ validatorAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryValidatorDelegationsRequest();
            request.setValidatorAddr(validatorAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.ValidatorDelegations);
                return transformers_1.ChainGrpcStakingTransformer.delegationsResponseToDelegations(response);
            }
            catch (e) {
                if (e.message.includes('does not exist')) {
                    return { delegations: [], pagination: { total: 0, next: '' } };
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchUnbondingDelegations({ injectiveAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegatorUnbondingDelegationsRequest();
            request.setDelegatorAddr(injectiveAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.DelegatorUnbondingDelegations);
                return transformers_1.ChainGrpcStakingTransformer.unBondingDelegationsResponseToUnBondingDelegations(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchUnbondingDelegationsNoThrow({ injectiveAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryDelegatorUnbondingDelegationsRequest();
            request.setDelegatorAddr(injectiveAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.DelegatorUnbondingDelegations);
                return transformers_1.ChainGrpcStakingTransformer.unBondingDelegationsResponseToUnBondingDelegations(response);
            }
            catch (e) {
                if (e.message.includes('does not exist')) {
                    return { unbondingDelegations: [], pagination: { total: 0, next: '' } };
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchReDelegations({ injectiveAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryRedelegationsRequest();
            request.setDelegatorAddr(injectiveAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Redelegations);
                return transformers_1.ChainGrpcStakingTransformer.reDelegationsResponseToReDelegations(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchReDelegationsNoThrow({ injectiveAddress, pagination, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryRedelegationsRequest();
            request.setDelegatorAddr(injectiveAddress);
            const paginationForRequest = (0, pagination_1.paginationRequestFromPagination)(pagination);
            if (paginationForRequest) {
                request.setPagination(paginationForRequest);
            }
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Redelegations);
                return transformers_1.ChainGrpcStakingTransformer.reDelegationsResponseToReDelegations(response);
            }
            catch (e) {
                if (e.message.includes('does not exist')) {
                    return { redelegations: [], pagination: { total: 0, next: '' } };
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.ChainGrpcStakingApi = ChainGrpcStakingApi;
