// @injectivelabs/sdk-ts/dist/client/chain/grpc/ChainGrpcMintApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcMintApi = void 0;
const query_pb_service_1 = require("@injectivelabs/chain-api/cosmos/mint/v1beta1/query_pb_service");
const query_pb_1 = require("@injectivelabs/chain-api/cosmos/mint/v1beta1/query_pb");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const utils_1 = require("../../../utils");
const utils_2 = require("@injectivelabs/utils");
const ChainGrpcMintTransformer_1 = require("./../transformers/ChainGrpcMintTransformer");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Chain Grpc API
 */
class ChainGrpcMintApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.ChainModule.Mint;
    }
    fetchModuleParams() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryParamsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Params);
                return ChainGrpcMintTransformer_1.ChainGrpcMintTransformer.moduleParamsResponseToModuleParams(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchInflation() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryInflationRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.Inflation);
                return {
                    inflation: (0, utils_1.cosmosSdkDecToBigNumber)(new utils_2.BigNumberInBase((0, utils_1.uint8ArrayToString)(response.getInflation()))).toFixed(),
                };
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchAnnualProvisions() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryAnnualProvisionsRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.AnnualProvisions);
                return {
                    annualProvisions: (0, utils_1.cosmosSdkDecToBigNumber)(new utils_2.BigNumberInBase((0, utils_1.uint8ArrayToString)(response.getAnnualProvisions()))).toFixed(),
                };
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.ChainGrpcMintApi = ChainGrpcMintApi;
