// @injectivelabs/sdk-ts/dist/client/chain/grpc/ChainGrpcAuctionApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcAuctionApi = void 0;
const query_pb_service_1 = require("@injectivelabs/chain-api/injective/auction/v1beta1//query_pb_service");
const query_pb_1 = require("@injectivelabs/chain-api/injective/auction/v1beta1/query_pb");
const exceptions_1 = require("@injectivelabs/exceptions");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const transformers_1 = require("../transformers");
const types_1 = require("../types");
/**
 * @category Chain Grpc API
 */
class ChainGrpcAuctionApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.ChainModule.Auction;
    }
    fetchModuleParams() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryAuctionParamsResponse();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.AuctionParams);
                return transformers_1.ChainGrpcAuctionTransformer.moduleParamsResponseToModuleParams(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchModuleState() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryModuleStateRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.AuctionModuleState);
                return transformers_1.ChainGrpcAuctionTransformer.auctionModuleStateResponseToAuctionModuleState(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchCurrentBasket() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new query_pb_1.QueryCurrentAuctionBasketRequest();
            try {
                const response = yield this.request(request, query_pb_service_1.Query.CurrentAuctionBasket);
                return transformers_1.ChainGrpcAuctionTransformer.currentBasketResponseToCurrentBasket(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.ChainGrpcAuctionApi = ChainGrpcAuctionApi;
