// @injectivelabs/sdk-ts/dist/client/indexer/grpc/IndexerGrpcInsuranceFundApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexerGrpcInsuranceFundApi = void 0;
const injective_insurance_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_insurance_rpc_pb");
const injective_insurance_rpc_pb_service_1 = require("@injectivelabs/indexer-api/injective_insurance_rpc_pb_service");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const transformers_1 = require("../transformers");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Indexer Grpc API
 */
class IndexerGrpcInsuranceFundApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.IndexerModule.InsuranceFund;
    }
    fetchRedemptions({ denom, address, status, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_insurance_rpc_pb_1.RedemptionsRequest();
            request.setRedeemer(address);
            if (denom) {
                request.setRedemptionDenom(denom);
            }
            if (status) {
                request.setStatus(status);
            }
            try {
                const response = yield this.request(request, injective_insurance_rpc_pb_service_1.InjectiveInsuranceRPC.Redemptions);
                return transformers_1.IndexerGrpcInsuranceFundTransformer.redemptionsResponseToRedemptions(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchInsuranceFunds() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_insurance_rpc_pb_1.FundsRequest();
            try {
                const response = yield this.request(request, injective_insurance_rpc_pb_service_1.InjectiveInsuranceRPC.Funds);
                return transformers_1.IndexerGrpcInsuranceFundTransformer.insuranceFundsResponseToInsuranceFunds(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.IndexerGrpcInsuranceFundApi = IndexerGrpcInsuranceFundApi;
