// @injectivelabs/sdk-ts/dist/client/indexer/grpc/IndexerGrpcMetaApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexerGrpcMetaApi = void 0;
const injective_meta_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_meta_rpc_pb");
const injective_meta_rpc_pb_service_1 = require("@injectivelabs/indexer-api/injective_meta_rpc_pb_service");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Indexer Grpc API
 */
class IndexerGrpcMetaApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.IndexerModule.Meta;
    }
    fetchPing() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_meta_rpc_pb_1.PingRequest();
            try {
                const response = yield this.request(request, injective_meta_rpc_pb_service_1.InjectiveMetaRPC.Ping);
                return response.toObject();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchVersion() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_meta_rpc_pb_1.VersionRequest();
            try {
                const response = yield this.request(request, injective_meta_rpc_pb_service_1.InjectiveMetaRPC.Version);
                return response.toObject();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchInfo() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_meta_rpc_pb_1.InfoRequest();
            try {
                const response = yield this.request(request, injective_meta_rpc_pb_service_1.InjectiveMetaRPC.Info);
                return response.toObject();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.IndexerGrpcMetaApi = IndexerGrpcMetaApi;
