// @injectivelabs/sdk-ts/dist/client/indexer/grpc/IndexerGrpcPortfolioApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexerGrpcAccountPortfolioApi = void 0;
const injective_portfolio_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_portfolio_rpc_pb");
const injective_portfolio_rpc_pb_service_1 = require("@injectivelabs/indexer-api/injective_portfolio_rpc_pb_service");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const transformers_1 = require("../transformers");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Indexer Grpc API
 */
class IndexerGrpcAccountPortfolioApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.IndexerModule.Portfolio;
    }
    fetchAccountPortfolio(address) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_portfolio_rpc_pb_1.AccountPortfolioRequest();
            request.setAccountAddress(address);
            try {
                const response = yield this.request(request, injective_portfolio_rpc_pb_service_1.InjectivePortfolioRPC.AccountPortfolio);
                return transformers_1.IndexerGrpcAccountPortfolioTransformer.accountPortfolioResponseToAccountPortfolio(response, address);
            }
            catch (e) {
                if ((e === null || e === void 0 ? void 0 : e.message) === 'account address not found') {
                    return {
                        accountAddress: address || '',
                        bankBalancesList: [],
                        subaccountsList: [],
                        positionsWithUpnlList: [],
                    };
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.IndexerGrpcAccountPortfolioApi = IndexerGrpcAccountPortfolioApi;
