// @injectivelabs/sdk-ts/dist/client/indexer/grpc/IndexerGrpcOracleApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexerGrpcOracleApi = void 0;
const injective_oracle_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_oracle_rpc_pb");
const injective_oracle_rpc_pb_service_1 = require("@injectivelabs/indexer-api/injective_oracle_rpc_pb_service");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const IndexerGrpcOracleTransformer_1 = require("../transformers/IndexerGrpcOracleTransformer");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Indexer Grpc API
 */
class IndexerGrpcOracleApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.IndexerModule.Oracle;
    }
    fetchOracleList() {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_oracle_rpc_pb_1.OracleListRequest();
            try {
                const response = yield this.request(request, injective_oracle_rpc_pb_service_1.InjectiveOracleRPC.OracleList);
                return IndexerGrpcOracleTransformer_1.IndexerGrpcOracleTransformer.oraclesResponseToOracles(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchOraclePrice({ baseSymbol, quoteSymbol, oracleScaleFactor, oracleType, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_oracle_rpc_pb_1.PriceRequest();
            request.setBaseSymbol(baseSymbol);
            request.setQuoteSymbol(quoteSymbol);
            request.setOracleType(oracleType);
            if (oracleScaleFactor) {
                request.setOracleScaleFactor(oracleScaleFactor);
            }
            try {
                const response = yield this.request(request, injective_oracle_rpc_pb_service_1.InjectiveOracleRPC.Price);
                return response.toObject();
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchOraclePriceNoThrow({ baseSymbol, quoteSymbol, oracleScaleFactor, oracleType, }) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_oracle_rpc_pb_1.PriceRequest();
            request.setBaseSymbol(baseSymbol);
            request.setQuoteSymbol(quoteSymbol);
            request.setOracleType(oracleType);
            if (oracleScaleFactor) {
                request.setOracleScaleFactor(oracleScaleFactor);
            }
            try {
                const response = yield this.request(request, injective_oracle_rpc_pb_service_1.InjectiveOracleRPC.Price);
                return response.toObject();
            }
            catch (e) {
                if (e.message.includes('object not found')) {
                    return {
                        price: '0',
                    };
                }
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.IndexerGrpcOracleApi = IndexerGrpcOracleApi;
