// @injectivelabs/sdk-ts/dist/client/indexer/grpc/IndexerGrpcSpotApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexerGrpcSpotApi = void 0;
const injective_spot_exchange_rpc_pb_service_1 = require("@injectivelabs/indexer-api/injective_spot_exchange_rpc_pb_service");
const injective_spot_exchange_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_spot_exchange_rpc_pb");
const BaseGrpcConsumer_1 = __importDefault(require("../../BaseGrpcConsumer"));
const transformers_1 = require("../transformers");
const types_1 = require("../types");
const exceptions_1 = require("@injectivelabs/exceptions");
/**
 * @category Indexer Grpc API
 */
class IndexerGrpcSpotApi extends BaseGrpcConsumer_1.default {
    constructor() {
        super(...arguments);
        this.module = types_1.IndexerModule.Spot;
    }
    fetchMarkets(params) {
        return __awaiter(this, void 0, void 0, function* () {
            const { baseDenom, marketStatus, quoteDenom } = params || {};
            const request = new injective_spot_exchange_rpc_pb_1.MarketsRequest();
            if (baseDenom) {
                request.setBaseDenom(baseDenom);
            }
            if (marketStatus) {
                request.setMarketStatus(marketStatus);
            }
            if (quoteDenom) {
                request.setQuoteDenom(quoteDenom);
            }
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.Markets);
                return transformers_1.IndexerGrpcSpotTransformer.marketsResponseToMarkets(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchMarket(marketId) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_spot_exchange_rpc_pb_1.MarketRequest();
            request.setMarketId(marketId);
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.Market);
                return transformers_1.IndexerGrpcSpotTransformer.marketResponseToMarket(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchOrderbook(marketId) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_spot_exchange_rpc_pb_1.OrderbookRequest();
            request.setMarketId(marketId);
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.Orderbook);
                return transformers_1.IndexerGrpcSpotTransformer.orderbookResponseToOrderbook(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchOrders(params) {
        return __awaiter(this, void 0, void 0, function* () {
            const { marketId, marketIds, subaccountId, orderSide, pagination } = params || {};
            const request = new injective_spot_exchange_rpc_pb_1.OrdersRequest();
            if (marketId) {
                request.setMarketId(marketId);
            }
            if (marketIds) {
                request.setMarketIdsList(marketIds);
            }
            if (subaccountId) {
                request.setSubaccountId(subaccountId);
            }
            if (orderSide) {
                request.setOrderSide(orderSide);
            }
            /*
            if (isConditional !== undefined) {
              request.setIsConditional(isConditional)
            }
            */
            if (pagination) {
                if (pagination.skip !== undefined) {
                    request.setSkip(pagination.skip);
                }
                if (pagination.limit !== undefined) {
                    request.setLimit(pagination.limit);
                }
                if (pagination.endTime !== undefined) {
                    request.setEndTime(pagination.endTime);
                }
            }
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.Orders);
                return transformers_1.IndexerGrpcSpotTransformer.ordersResponseToOrders(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchOrderHistory(params) {
        return __awaiter(this, void 0, void 0, function* () {
            const { subaccountId, marketId, marketIds, orderTypes, executionTypes, direction, state, pagination, } = params || {};
            const request = new injective_spot_exchange_rpc_pb_1.OrdersHistoryRequest();
            if (subaccountId) {
                request.setSubaccountId(subaccountId);
            }
            if (marketId) {
                request.setMarketId(marketId);
            }
            if (marketIds) {
                request.setMarketIdsList(marketIds);
            }
            if (orderTypes) {
                request.setOrderTypesList(orderTypes);
            }
            if (executionTypes) {
                request.setExecutionTypesList(executionTypes);
            }
            if (direction) {
                request.setDirection(direction);
            }
            if (state) {
                request.setState(state);
            }
            /*
            if (isConditional !== undefined) {
              request.setIsConditional(isConditional)
            }
            */
            if (pagination) {
                if (pagination.skip !== undefined) {
                    request.setSkip(pagination.skip);
                }
                if (pagination.limit !== undefined) {
                    request.setLimit(pagination.limit);
                }
                if (pagination.endTime !== undefined) {
                    request.setEndTime(pagination.endTime);
                }
            }
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.OrdersHistory);
                return transformers_1.IndexerGrpcSpotTransformer.orderHistoryResponseToOrderHistory(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchTrades(params) {
        return __awaiter(this, void 0, void 0, function* () {
            const { marketId, pagination, subaccountId, executionTypes, executionSide, startTime, endTime, direction, marketIds, } = params || {};
            const request = new injective_spot_exchange_rpc_pb_1.TradesRequest();
            if (marketId) {
                request.setMarketId(marketId);
            }
            if (marketIds) {
                request.setMarketIdsList(marketIds);
            }
            else {
                request.setMarketIdsList([]);
            }
            if (subaccountId) {
                request.setSubaccountId(subaccountId);
            }
            if (executionTypes) {
                request.setExecutionTypesList(executionTypes);
            }
            if (executionSide) {
                request.setExecutionSide(executionSide);
            }
            if (direction) {
                request.setDirection(direction);
            }
            if (startTime) {
                request.setStartTime(startTime);
            }
            if (endTime) {
                request.setEndTime(endTime);
            }
            if (pagination) {
                if (pagination.skip !== undefined) {
                    request.setSkip(pagination.skip);
                }
                if (pagination.limit !== undefined) {
                    request.setLimit(pagination.limit);
                }
                if (pagination.endTime !== undefined) {
                    request.setEndTime(pagination.endTime);
                }
            }
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.Trades);
                return transformers_1.IndexerGrpcSpotTransformer.tradesResponseToTrades(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchSubaccountOrdersList(params) {
        return __awaiter(this, void 0, void 0, function* () {
            const { subaccountId, marketId, pagination } = params || {};
            const request = new injective_spot_exchange_rpc_pb_1.SubaccountOrdersListRequest();
            if (subaccountId) {
                request.setSubaccountId(subaccountId);
            }
            if (marketId) {
                request.setMarketId(marketId);
            }
            if (pagination) {
                if (pagination.skip !== undefined) {
                    request.setSkip(pagination.skip);
                }
                if (pagination.limit !== undefined) {
                    request.setLimit(pagination.limit);
                }
            }
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.SubaccountOrdersList);
                return transformers_1.IndexerGrpcSpotTransformer.ordersResponseToOrders(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchSubaccountTradesList(params) {
        return __awaiter(this, void 0, void 0, function* () {
            const { subaccountId, marketId, direction, executionType, pagination } = params || {};
            const request = new injective_spot_exchange_rpc_pb_1.SubaccountTradesListRequest();
            if (subaccountId) {
                request.setSubaccountId(subaccountId);
            }
            if (marketId) {
                request.setMarketId(marketId);
            }
            if (direction) {
                request.setDirection(direction);
            }
            if (executionType) {
                request.setExecutionType(executionType);
            }
            if (pagination) {
                if (pagination.skip !== undefined) {
                    request.setSkip(pagination.skip);
                }
                if (pagination.limit !== undefined) {
                    request.setLimit(pagination.limit);
                }
            }
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.SubaccountTradesList);
                return transformers_1.IndexerGrpcSpotTransformer.subaccountTradesListResponseToTradesList(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchOrderbooks(marketIds) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_spot_exchange_rpc_pb_1.OrderbooksRequest();
            if (marketIds.length > 0) {
                request.setMarketIdsList(marketIds);
            }
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.Orderbooks);
                return transformers_1.IndexerGrpcSpotTransformer.orderbooksResponseToOrderbooks(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchOrderbooksV2(marketIds) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_spot_exchange_rpc_pb_1.OrderbooksV2Request();
            if (marketIds.length > 0) {
                request.setMarketIdsList(marketIds);
            }
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.OrderbooksV2);
                return transformers_1.IndexerGrpcSpotTransformer.orderbooksV2ResponseToOrderbooksV2(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
    fetchOrderbookV2(marketId) {
        return __awaiter(this, void 0, void 0, function* () {
            const request = new injective_spot_exchange_rpc_pb_1.OrderbookV2Request();
            request.setMarketId(marketId);
            try {
                const response = yield this.request(request, injective_spot_exchange_rpc_pb_service_1.InjectiveSpotExchangeRPC.OrderbookV2);
                return transformers_1.IndexerGrpcSpotTransformer.orderbookV2ResponseToOrderbookV2(response);
            }
            catch (e) {
                if (e instanceof exceptions_1.GrpcUnaryRequestException) {
                    throw e;
                }
                throw new exceptions_1.GrpcUnaryRequestException(e, {
                    code: exceptions_1.UnspecifiedErrorCode,
                    contextModule: this.module,
                });
            }
        });
    }
}
exports.IndexerGrpcSpotApi = IndexerGrpcSpotApi;
