// @injectivelabs/sdk-ts/dist/core/modules/gov/msgs/MsgDeposit.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/cosmos/gov/v1beta1/tx_pb");
const coin_pb_1 = require("@injectivelabs/chain-api/cosmos/base/v1beta1/coin_pb");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
const MsgBase_1 = require("../../MsgBase");
/**
 * @category Messages
 */
class MsgDeposit extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgDeposit(params);
    }
    toProto() {
        const { params } = this;
        const deposit = new coin_pb_1.Coin();
        deposit.setAmount(params.amount.amount);
        deposit.setDenom(params.amount.denom);
        const message = new tx_pb_1.MsgDeposit();
        message.setDepositor(params.depositor);
        message.setProposalId(params.proposalId);
        message.setAmountList([deposit]);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmos.gov.v1beta1.MsgDeposit' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign(Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject())), { amount: proto
                .getAmountList()
                .map((amount) => (0, snakecase_keys_1.default)(amount.toObject())) });
        // @ts-ignore
        delete message.amount_list;
        return {
            type: 'cosmos-sdk/MsgDeposit',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmos.gov.v1beta1.MsgDeposit' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmos.gov.v1beta1.MsgDeposit',
            message: proto,
        };
    }
}
exports.default = MsgDeposit;
