// @injectivelabs/chain-api/ibc/core/port/v1/query_pb_service.js
// package: ibc.core.port.v1
// file: ibc/core/port/v1/query.proto

var ibc_core_port_v1_query_pb = require("../../../../ibc/core/port/v1/query_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Query = (function () {
  function Query() {}
  Query.serviceName = "ibc.core.port.v1.Query";
  return Query;
}());

Query.AppVersion = {
  methodName: "AppVersion",
  service: Query,
  requestStream: false,
  responseStream: false,
  requestType: ibc_core_port_v1_query_pb.QueryAppVersionRequest,
  responseType: ibc_core_port_v1_query_pb.QueryAppVersionResponse
};

exports.Query = Query;

function QueryClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

QueryClient.prototype.appVersion = function appVersion(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Query.AppVersion, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.QueryClient = QueryClient;
