// @injectivelabs/ts-types/dist/cjs/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainId = exports.EthereumChainId = void 0;
var EthereumChainId;
(function (EthereumChainId) {
    EthereumChainId[EthereumChainId["Mainnet"] = 1] = "Mainnet";
    EthereumChainId[EthereumChainId["Ropsten"] = 3] = "Ropsten";
    EthereumChainId[EthereumChainId["Rinkeby"] = 4] = "Rinkeby";
    EthereumChainId[EthereumChainId["Goerli"] = 5] = "Goerli";
    EthereumChainId[EthereumChainId["Kovan"] = 42] = "Kovan";
    EthereumChainId[EthereumChainId["Injective"] = 888] = "Injective";
    EthereumChainId[EthereumChainId["Ganache"] = 1337] = "Ganache";
    EthereumChainId[EthereumChainId["HardHat"] = 31337] = "HardHat";
})(EthereumChainId = exports.EthereumChainId || (exports.EthereumChainId = {}));
var ChainId;
(function (ChainId) {
    ChainId["Mainnet"] = "injective-1";
    ChainId["Testnet"] = "injective-888";
    ChainId["Devnet"] = "injective-777";
})(ChainId = exports.ChainId || (exports.ChainId = {}));
