// @injectivelabs/ts-types/dist/cjs/trade.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderState = exports.TradeDirection = exports.TradeExecutionSide = exports.TradeExecutionType = void 0;
var TradeExecutionType;
(function (TradeExecutionType) {
    TradeExecutionType["Market"] = "market";
    TradeExecutionType["LimitFill"] = "limitFill";
    TradeExecutionType["LimitMatchRestingOrder"] = "limitMatchRestingOrder";
    TradeExecutionType["LimitMatchNewOrder"] = "limitMatchNewOrder";
})(TradeExecutionType = exports.TradeExecutionType || (exports.TradeExecutionType = {}));
var TradeExecutionSide;
(function (TradeExecutionSide) {
    TradeExecutionSide["Maker"] = "maker";
    TradeExecutionSide["Taker"] = "taker";
})(TradeExecutionSide = exports.TradeExecutionSide || (exports.TradeExecutionSide = {}));
var TradeDirection;
(function (TradeDirection) {
    TradeDirection["Buy"] = "buy";
    TradeDirection["Sell"] = "sell";
    TradeDirection["Long"] = "long";
    TradeDirection["Short"] = "short";
})(TradeDirection = exports.TradeDirection || (exports.TradeDirection = {}));
var OrderState;
(function (OrderState) {
    OrderState["Booked"] = "booked";
    OrderState["PartiallyFilled"] = "partially_filled";
    OrderState["Filled"] = "filled";
    OrderState["Canceled"] = "canceled";
    OrderState["Triggered"] = "triggered";
})(OrderState = exports.OrderState || (exports.OrderState = {}));
