// @injectivelabs/sdk-ts/dist/types/exchange.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderMask = exports.TradeDirection = exports.TradeExecutionSide = exports.TradeExecutionType = void 0;
var TradeExecutionType;
(function (TradeExecutionType) {
    TradeExecutionType["Market"] = "market";
    TradeExecutionType["LimitFill"] = "limitFill";
    TradeExecutionType["LimitMatchRestingOrder"] = "limitMatchRestingOrder";
    TradeExecutionType["LimitMatchNewOrder"] = "limitMatchNewOrder";
})(TradeExecutionType = exports.TradeExecutionType || (exports.TradeExecutionType = {}));
var TradeExecutionSide;
(function (TradeExecutionSide) {
    TradeExecutionSide["Maker"] = "maker";
    TradeExecutionSide["Taker"] = "taker";
})(TradeExecutionSide = exports.TradeExecutionSide || (exports.TradeExecutionSide = {}));
var TradeDirection;
(function (TradeDirection) {
    TradeDirection["Buy"] = "buy";
    TradeDirection["Sell"] = "sell";
    TradeDirection["Long"] = "long";
    TradeDirection["Short"] = "short";
})(TradeDirection = exports.TradeDirection || (exports.TradeDirection = {}));
var OrderMask;
(function (OrderMask) {
    OrderMask[OrderMask["Unused"] = 0] = "Unused";
    OrderMask[OrderMask["Any"] = 1] = "Any";
    OrderMask[OrderMask["Regular"] = 2] = "Regular";
    OrderMask[OrderMask["Conditional"] = 4] = "Conditional";
    OrderMask[OrderMask["DirectionBuyOrHigher"] = 8] = "DirectionBuyOrHigher";
    OrderMask[OrderMask["DirectionSellOrLower"] = 16] = "DirectionSellOrLower";
    OrderMask[OrderMask["TypeMarket"] = 32] = "TypeMarket";
    OrderMask[OrderMask["TypeLimit"] = 64] = "TypeLimit";
})(OrderMask = exports.OrderMask || (exports.OrderMask = {}));
