// @injectivelabs/sdk-ts/dist/core/modules/wasm/msgs/MsgExecuteContract.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const coin_pb_1 = require("@injectivelabs/chain-api/cosmos/base/v1beta1/coin_pb");
const tx_pb_1 = require("@injectivelabs/chain-api/cosmwasm/wasm/v1/tx_pb");
const MsgBase_1 = require("../../MsgBase");
const utf8_1 = require("../../../../utils/utf8");
const exceptions_1 = require("@injectivelabs/exceptions");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgExecuteContract extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgExecuteContract(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgExecuteContract();
        const msg = this.getMsgObject();
        message.setMsg((0, utf8_1.fromUtf8)(JSON.stringify(msg)));
        message.setSender(params.sender);
        message.setContract(params.contractAddress);
        if (params.funds) {
            const fundsToArray = Array.isArray(params.funds)
                ? params.funds
                : [params.funds];
            const funds = fundsToArray.map((coin) => {
                const funds = new coin_pb_1.Coin();
                funds.setAmount(coin.amount);
                funds.setDenom(coin.denom);
                return funds;
            });
            message.setFundsList(funds);
        }
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgExecuteContract' }, proto.toObject());
    }
    toAmino() {
        const { params } = this;
        const proto = this.toProto();
        const funds = params.funds && {
            funds: proto
                .getFundsList()
                .map((amount) => (0, snakecase_keys_1.default)(amount.toObject())),
        };
        const message = Object.assign(Object.assign(Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject())), funds), { msg: this.getMsgObject() });
        // @ts-ignore
        delete message.funds_list;
        return {
            type: 'wasm/MsgExecuteContract',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgExecuteContract' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmwasm.wasm.v1.MsgExecuteContract',
            message: proto,
        };
    }
    getMsgObject() {
        const { params } = this;
        if ((params.exec || params.msg) && params.execArgs) {
            throw new exceptions_1.GeneralException(new Error('Please provide only one exec|msg argument'));
        }
        if (params.execArgs) {
            return params.execArgs.toExecData();
        }
        if (params.exec) {
            return {
                [params.exec.action]: params.exec.msg,
            };
        }
        if (params.msg) {
            return params.msg;
        }
        throw new exceptions_1.GeneralException(new Error('Please provide at least one exec argument'));
    }
}
exports.default = MsgExecuteContract;
