// @injectivelabs/sdk-ts/dist/core/modules/wasm/msgs/MsgExecuteContractCompat.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/injective/wasmx/v1/tx_pb");
const MsgBase_1 = require("../../MsgBase");
const exceptions_1 = require("@injectivelabs/exceptions");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgExecuteContractCompat extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgExecuteContractCompat(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgExecuteContractCompat();
        const msg = this.getMsgObject();
        message.setMsg(JSON.stringify(msg));
        message.setSender(params.sender);
        message.setContract(params.contractAddress);
        if (params.funds) {
            const fundsToArray = Array.isArray(params.funds)
                ? params.funds
                : [params.funds];
            const funds = fundsToArray.map((coin) => {
                return `${coin.amount}${coin.denom}`;
            });
            message.setFunds(funds.join(','));
        }
        else {
            message.setFunds('0');
        }
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/injective.wasmx.v1.MsgExecuteContractCompat' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign(Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject())), { msg: JSON.stringify(this.getMsgObject()) });
        // @ts-ignore
        delete message.funds_list;
        return {
            type: 'wasmx/MsgExecuteContractCompat',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/injective.wasmx.v1.MsgExecuteContractCompat' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/injective.wasmx.v1.MsgExecuteContractCompat',
            message: proto,
        };
    }
    getMsgObject() {
        const { params } = this;
        if ((params.exec || params.msg) && params.execArgs) {
            throw new exceptions_1.GeneralException(new Error('Please provide only one exec|msg argument'));
        }
        if (params.execArgs) {
            return params.execArgs.toExecData();
        }
        if (params.exec) {
            return {
                [params.exec.action]: params.exec.msg,
            };
        }
        if (params.msg) {
            return params.msg;
        }
        throw new exceptions_1.GeneralException(new Error('Please provide at least one exec argument'));
    }
}
exports.default = MsgExecuteContractCompat;
