// @injectivelabs/sdk-ts/dist/core/modules/wasm/msgs/MsgInstantiateContract.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const coin_pb_1 = require("@injectivelabs/chain-api/cosmos/base/v1beta1/coin_pb");
const tx_pb_1 = require("@injectivelabs/chain-api/cosmwasm/wasm/v1/tx_pb");
const utils_1 = require("../../../../utils");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgInstantiateContract extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgInstantiateContract(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgInstantiateContract();
        message.setMsg((0, utils_1.fromUtf8)(JSON.stringify(params.msg)));
        message.setSender(params.sender);
        message.setAdmin(params.admin);
        message.setCodeId(params.codeId);
        message.setLabel(params.label);
        if (params.amount) {
            const funds = new coin_pb_1.Coin();
            funds.setAmount(params.amount.amount);
            funds.setDenom(params.amount.denom);
            message.setFundsList([funds]);
        }
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgInstantiateContract' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        // @ts-ignore
        delete message.funds_list;
        return {
            type: 'wasm/MsgInstantiateContract',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgInstantiateContract' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmwasm.wasm.v1.MsgInstantiateContract',
            message: proto,
        };
    }
}
exports.default = MsgInstantiateContract;
