// @injectivelabs/sdk-ts/dist/core/modules/wasm/msgs/MsgMigrateContract.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/cosmwasm/wasm/v1/tx_pb");
const utils_1 = require("../../../../utils");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgMigrateContract extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgMigrateContract(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgMigrateContract();
        message.setCodeId(params.codeId);
        message.setContract(params.contract);
        message.setSender(params.sender);
        message.setMsg((0, utils_1.fromUtf8)(JSON.stringify(params.msg)));
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgMigrateContract' }, proto.toObject());
    }
    toAmino() {
        const { params } = this;
        const proto = this.toProto();
        const message = Object.assign(Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject())), { msg: params.msg });
        return {
            type: 'wasm/MsgMigrateContract',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgMigrateContract' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmwasm.wasm.v1.MsgMigrateContract',
            message: proto,
        };
    }
}
exports.default = MsgMigrateContract;
