// @injectivelabs/sdk-ts/dist/core/modules/wasm/msgs/MsgStoreCode.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/cosmwasm/wasm/v1/tx_pb");
const utils_1 = require("../../../../utils");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgStoreCode extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgStoreCode(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgStoreCode();
        message.setSender(params.sender);
        message.setWasmByteCode(typeof params.wasmBytes === 'string'
            ? (0, utils_1.fromUtf8)(params.wasmBytes)
            : params.wasmBytes);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgStoreCode' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'wasm/MsgStoreCode',
            value: Object.assign({}, message),
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgStoreCode' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmwasm.wasm.v1.MsgStoreCode',
            message: proto,
        };
    }
}
exports.default = MsgStoreCode;
