// @injectivelabs/sdk-ts/dist/core/modules/wasm/msgs/MsgUpdateAdmin.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/cosmwasm/wasm/v1/tx_pb");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgUpdateAdmin extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgUpdateAdmin(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgUpdateAdmin();
        message.setSender(params.sender);
        message.setNewAdmin(params.newAdmin);
        message.setContract(params.contract);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgUpdateAdmin' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'wasm/MsgUpdateAdmin',
            value: Object.assign({}, message),
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmwasm.wasm.v1.MsgUpdateAdmin' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmwasm.wasm.v1.MsgUpdateAdmin',
            message: proto,
        };
    }
}
exports.default = MsgUpdateAdmin;
