// @injectivelabs/sdk-ts/dist/client/indexer/grpc_stream/IndexerGrpcDerivativesStream.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexerGrpcDerivativesStream = void 0;
const injective_derivative_exchange_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_derivative_exchange_rpc_pb");
const injective_derivative_exchange_rpc_pb_service_1 = require("@injectivelabs/indexer-api/injective_derivative_exchange_rpc_pb_service");
const transformers_1 = require("../transformers");
const grpc_1 = require("../../../utils/grpc");
/**
 * @category Indexer Grpc Stream
 */
class IndexerGrpcDerivativesStream {
    constructor(endpoint) {
        this.client = new injective_derivative_exchange_rpc_pb_service_1.InjectiveDerivativeExchangeRPCClient(endpoint, {
            transport: (0, grpc_1.getGrpcTransport)(),
        });
    }
    streamDerivativeOrderbook({ marketIds, callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_derivative_exchange_rpc_pb_1.StreamOrderbookRequest();
        request.setMarketIdsList(marketIds);
        const stream = this.client.streamOrderbook(request);
        stream.on('data', (response) => {
            callback(transformers_1.IndexerDerivativeStreamTransformer.orderbookStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    streamDerivativeOrders({ marketId, subaccountId, orderSide, callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_derivative_exchange_rpc_pb_1.StreamOrdersRequest();
        if (marketId) {
            request.setMarketId(marketId);
        }
        if (subaccountId) {
            request.setSubaccountId(subaccountId);
        }
        if (orderSide) {
            request.setOrderSide(orderSide);
        }
        const stream = this.client.streamOrders(request);
        stream.on('data', (response) => {
            callback(transformers_1.IndexerDerivativeStreamTransformer.ordersStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    streamDerivativeOrderHistory({ subaccountId, marketId, orderTypes, executionTypes, direction, state, callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_derivative_exchange_rpc_pb_1.StreamOrdersHistoryRequest();
        if (subaccountId) {
            request.setSubaccountId(subaccountId);
        }
        if (marketId) {
            request.setMarketId(marketId);
        }
        if (orderTypes) {
            request.setOrderTypesList(orderTypes);
        }
        if (direction) {
            request.setDirection(direction);
        }
        if (state) {
            request.setState(state);
        }
        if (executionTypes) {
            request.setExecutionTypesList(executionTypes);
        }
        const stream = this.client.streamOrdersHistory(request);
        stream.on('data', (response) => {
            callback(transformers_1.IndexerDerivativeStreamTransformer.orderHistoryStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    streamDerivativeTrades({ marketIds, marketId, subaccountIds, subaccountId, callback, pagination, executionSide, direction, onEndCallback, onStatusCallback, }) {
        const request = new injective_derivative_exchange_rpc_pb_1.StreamTradesRequest();
        if (marketIds) {
            request.setMarketIdsList(marketIds);
        }
        if (marketId) {
            request.setMarketId(marketId);
        }
        if (subaccountIds) {
            request.setSubaccountIdsList(subaccountIds);
        }
        if (subaccountId) {
            request.setSubaccountId(subaccountId);
        }
        if (executionSide) {
            request.setExecutionSide(executionSide);
        }
        if (direction) {
            request.setDirection(direction);
        }
        if (pagination) {
            if (pagination.skip !== undefined) {
                request.setSkip(pagination.skip);
            }
            if (pagination.limit !== undefined) {
                request.setLimit(pagination.limit);
            }
        }
        const stream = this.client.streamTrades(request);
        stream.on('data', (response) => {
            callback(transformers_1.IndexerDerivativeStreamTransformer.tradesStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    streamDerivativePositions({ marketId, subaccountId, callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_derivative_exchange_rpc_pb_1.StreamPositionsRequest();
        if (marketId) {
            request.setMarketId(marketId);
        }
        if (subaccountId) {
            request.setSubaccountId(subaccountId);
        }
        const stream = this.client.streamPositions(request);
        stream.on('data', (response) => {
            callback(transformers_1.IndexerDerivativeStreamTransformer.positionStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    streamDerivativeMarket({ marketIds, callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_derivative_exchange_rpc_pb_1.StreamMarketRequest();
        if (marketIds) {
            request.setMarketIdsList(marketIds);
        }
        const stream = this.client.streamMarket(request);
        stream.on('data', (response) => {
            callback(response);
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    streamDerivativeOrderbookV2({ marketIds, callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_derivative_exchange_rpc_pb_1.StreamOrderbookV2Request();
        request.setMarketIdsList(marketIds);
        const stream = this.client.streamOrderbookV2(request);
        stream.on('data', (response) => {
            callback(transformers_1.IndexerDerivativeStreamTransformer.orderbookV2StreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    streamDerivativeOrderbookUpdate({ marketIds, callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_derivative_exchange_rpc_pb_1.StreamOrderbookUpdateRequest();
        request.setMarketIdsList(marketIds);
        const stream = this.client.streamOrderbookUpdate(request);
        stream.on('data', (response) => {
            callback(transformers_1.IndexerDerivativeStreamTransformer.orderbookUpdateStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
}
exports.IndexerGrpcDerivativesStream = IndexerGrpcDerivativesStream;
