// @injectivelabs/sdk-ts/dist/client/indexer/grpc_stream/IndexerGrpcExplorerStream.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexerGrpcExplorerStream = void 0;
const injective_explorer_rpc_pb_service_1 = require("@injectivelabs/indexer-api/injective_explorer_rpc_pb_service");
const injective_explorer_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_explorer_rpc_pb");
const transformers_1 = require("../transformers");
const grpc_1 = require("../../../utils/grpc");
/**
 * @category Indexer Grpc Stream
 */
class IndexerGrpcExplorerStream {
    constructor(endpoint) {
        this.client = new injective_explorer_rpc_pb_service_1.InjectiveExplorerRPCClient(endpoint, {
            transport: (0, grpc_1.getGrpcTransport)(),
        });
    }
    blocks({ callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_explorer_rpc_pb_1.StreamBlocksRequest();
        const stream = this.client.streamBlocks(request);
        stream.on('data', (response) => {
            callback(transformers_1.ExplorerStreamTransformer.blocksStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    blocksWithTxs({ callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_explorer_rpc_pb_1.StreamBlocksRequest();
        const stream = this.client.streamBlocks(request);
        stream.on('data', (response) => {
            callback(transformers_1.ExplorerStreamTransformer.blocksWithTxsStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
    streamTransactions({ callback, onEndCallback, onStatusCallback, }) {
        const request = new injective_explorer_rpc_pb_1.StreamTxsRequest();
        const stream = this.client.streamTxs(request);
        stream.on('data', (response) => {
            callback(transformers_1.ExplorerStreamTransformer.transactionsStreamCallback(response));
        });
        if (onEndCallback) {
            stream.on('end', onEndCallback);
        }
        if (onStatusCallback) {
            stream.on('status', onStatusCallback);
        }
        return stream;
    }
}
exports.IndexerGrpcExplorerStream = IndexerGrpcExplorerStream;
