// @injectivelabs/sdk-ts/dist/core/modules/gov/ProposalContentDecomposer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProposalDecomposer = void 0;
const proposal_pb_1 = require("@injectivelabs/chain-api/injective/oracle/v1beta1/proposal_pb");
const gov_pb_1 = require("@injectivelabs/chain-api/cosmos/gov/v1beta1/gov_pb");
const upgrade_pb_1 = require("@injectivelabs/chain-api/cosmos/upgrade/v1beta1/upgrade_pb");
const params_pb_1 = require("@injectivelabs/chain-api/cosmos/params/v1beta1/params_pb");
const tx_pb_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/tx_pb");
class ProposalDecomposer {
    static grantBandOraclePrivilegeProposal(content) {
        return proposal_pb_1.GrantBandOraclePrivilegeProposal.deserializeBinary(content);
    }
    static removeBandOraclePrivilegeProposal(content) {
        return proposal_pb_1.RevokeBandOraclePrivilegeProposal.deserializeBinary(content);
    }
    static grantPriceFeederPrivilegeProposal(content) {
        return proposal_pb_1.GrantPriceFeederPrivilegeProposal.deserializeBinary(content);
    }
    static removePriceFeederPrivilegeProposal(content) {
        return proposal_pb_1.RevokePriceFeederPrivilegeProposal.deserializeBinary(content);
    }
    static textProposal(content) {
        return gov_pb_1.TextProposal.deserializeBinary(content);
    }
    static SoftwareUpgrade(content) {
        return upgrade_pb_1.SoftwareUpgradeProposal.deserializeBinary(content);
    }
    static spotMarketLaunch(content) {
        return tx_pb_1.SpotMarketLaunchProposal.deserializeBinary(content);
    }
    static exchangeEnableProposal(content) {
        return tx_pb_1.ExchangeEnableProposal.deserializeBinary(content);
    }
    static spotMarketUpdate(content) {
        return tx_pb_1.SpotMarketParamUpdateProposal.deserializeBinary(content);
    }
    static perpetualMarketLaunch(content) {
        return tx_pb_1.PerpetualMarketLaunchProposal.deserializeBinary(content);
    }
    static expiryFuturesMarketLaunch(content) {
        return tx_pb_1.ExpiryFuturesMarketLaunchProposal.deserializeBinary(content);
    }
    static derivativeMarketUpdate(content) {
        return tx_pb_1.DerivativeMarketParamUpdateProposal.deserializeBinary(content);
    }
    static FeeDiscount(content) {
        return tx_pb_1.FeeDiscountProposal.deserializeBinary(content);
    }
    static TradingRewardCampaignLaunch(content) {
        return tx_pb_1.TradingRewardCampaignLaunchProposal.deserializeBinary(content);
    }
    static TradingRewardCampaignUpdate(content) {
        return tx_pb_1.TradingRewardCampaignUpdateProposal.deserializeBinary(content);
    }
    static parametersChange(content) {
        return params_pb_1.ParameterChangeProposal.deserializeBinary(content);
    }
    static EnableBandIBC(content) {
        return proposal_pb_1.EnableBandIBCProposal.deserializeBinary(content);
    }
    static AuthorizeBandOracleRequest(content) {
        return proposal_pb_1.AuthorizeBandOracleRequestProposal.deserializeBinary(content);
    }
}
exports.ProposalDecomposer = ProposalDecomposer;
