// @injectivelabs/sdk-ts/dist/client/chain/transformers/ChainGrpcDistributionTransformer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChainGrpcDistributionTransformer = void 0;
const utils_1 = require("../../../utils");
/**
 * @category Chain Grpc Transformer
 */
class ChainGrpcDistributionTransformer {
    static moduleParamsResponseToModuleParams(response) {
        const params = response.getParams();
        return {
            communityTax: (0, utils_1.cosmosSdkDecToBigNumber)(params.getCommunityTax()).toFixed(),
            baseProposerReward: params.getBaseProposerReward(),
            bonusProposerReward: params.getBonusProposerReward(),
            withdrawAddrEnabled: params.getWithdrawAddrEnabled(),
        };
    }
    static delegationRewardResponseToReward(response) {
        const grpcRewards = response.getRewardsList();
        return grpcRewards.map((grpcReward) => {
            return {
                amount: (0, utils_1.cosmosSdkDecToBigNumber)(grpcReward.getAmount()).toFixed(),
                denom: grpcReward.getDenom(),
            };
        });
    }
    static totalDelegationRewardResponseToTotalReward(response) {
        const grpcRewards = response.getRewardsList();
        return grpcRewards.map((grpcReward) => {
            const rewards = grpcReward.getRewardList().map((reward) => ({
                amount: (0, utils_1.cosmosSdkDecToBigNumber)(reward.getAmount()).toFixed(),
                denom: reward.getDenom(),
            }));
            return {
                rewards,
                validatorAddress: grpcReward.getValidatorAddress(),
            };
        });
    }
}
exports.ChainGrpcDistributionTransformer = ChainGrpcDistributionTransformer;
