// @injectivelabs/sdk-ts/dist/core/modules/distribution/msgs/MsgWithdrawDelegatorReward.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/cosmos/distribution/v1beta1/tx_pb");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgWithdrawDelegatorReward extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgWithdrawDelegatorReward(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgWithdrawDelegatorReward();
        message.setDelegatorAddress(params.delegatorAddress);
        message.setValidatorAddress(params.validatorAddress);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'cosmos-sdk/MsgWithdrawDelegationReward',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward',
            message: proto,
        };
    }
}
exports.default = MsgWithdrawDelegatorReward;
