// @injectivelabs/chain-api/injective/wasmx/v1/tx_pb_service.js
// package: injective.wasmx.v1
// file: injective/wasmx/v1/tx.proto

var injective_wasmx_v1_tx_pb = require("../../../injective/wasmx/v1/tx_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var Msg = (function () {
  function Msg() {}
  Msg.serviceName = "injective.wasmx.v1.Msg";
  return Msg;
}());

Msg.ExecuteContractCompat = {
  methodName: "ExecuteContractCompat",
  service: Msg,
  requestStream: false,
  responseStream: false,
  requestType: injective_wasmx_v1_tx_pb.MsgExecuteContractCompat,
  responseType: injective_wasmx_v1_tx_pb.MsgExecuteContractCompatResponse
};

exports.Msg = Msg;

function MsgClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

MsgClient.prototype.executeContractCompat = function executeContractCompat(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(Msg.ExecuteContractCompat, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

exports.MsgClient = MsgClient;
