// @injectivelabs/sdk-ts/dist/client/indexer/transformers/IndexerSpotStreamTransformer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexerSpotStreamTransformer = void 0;
const IndexerGrpcSpotTransformer_1 = require("./IndexerGrpcSpotTransformer");
/**
 * @category Indexer Stream Transformer
 */
class IndexerSpotStreamTransformer {
}
exports.IndexerSpotStreamTransformer = IndexerSpotStreamTransformer;
IndexerSpotStreamTransformer.orderbookStreamCallback = (response) => {
    const orderbook = response.getOrderbook();
    return {
        orderbook: orderbook
            ? IndexerGrpcSpotTransformer_1.IndexerGrpcSpotTransformer.grpcOrderbookToOrderbook({
                buys: orderbook.getBuysList(),
                sells: orderbook.getSellsList(),
            })
            : undefined,
        operation: response.getOperationType(),
        marketId: response.getMarketId(),
        timestamp: response.getTimestamp(),
    };
};
IndexerSpotStreamTransformer.tradesStreamCallback = (response) => {
    const trade = response.getTrade();
    return {
        trade: trade
            ? IndexerGrpcSpotTransformer_1.IndexerGrpcSpotTransformer.grpcTradeToTrade(trade)
            : undefined,
        operation: response.getOperationType(),
        timestamp: response.getTimestamp(),
    };
};
IndexerSpotStreamTransformer.ordersStreamCallback = (response) => {
    const order = response.getOrder();
    return {
        order: order
            ? IndexerGrpcSpotTransformer_1.IndexerGrpcSpotTransformer.grpcOrderToOrder(order)
            : undefined,
        operation: response.getOperationType(),
        timestamp: response.getTimestamp(),
    };
};
IndexerSpotStreamTransformer.orderHistoryStreamCallback = (response) => {
    const order = response.getOrder();
    return {
        order: order
            ? IndexerGrpcSpotTransformer_1.IndexerGrpcSpotTransformer.grpcOrderHistoryToOrderHistory(order)
            : undefined,
        operation: response.getOperationType(),
        timestamp: response.getTimestamp(),
    };
};
IndexerSpotStreamTransformer.orderbookV2StreamCallback = (response) => {
    const orderbook = response.getOrderbook();
    return {
        orderbook: orderbook
            ? IndexerGrpcSpotTransformer_1.IndexerGrpcSpotTransformer.grpcOrderbookV2ToOrderbookV2({
                sequence: orderbook.getSequence(),
                buys: orderbook.getBuysList(),
                sells: orderbook.getSellsList(),
            })
            : undefined,
        operation: response.getOperationType(),
        marketId: response.getMarketId(),
        timestamp: response.getTimestamp(),
    };
};
IndexerSpotStreamTransformer.orderbookUpdateStreamCallback = (response) => {
    const orderbook = response.getOrderbookLevelUpdates();
    return {
        orderbook: orderbook
            ? IndexerGrpcSpotTransformer_1.IndexerGrpcSpotTransformer.grpcOrderbookV2ToOrderbookV2({
                sequence: orderbook.getSequence(),
                buys: orderbook.getBuysList(),
                sells: orderbook.getSellsList(),
            })
            : undefined,
        operation: response.getOperationType(),
        marketId: response.getMarketId(),
        timestamp: response.getTimestamp(),
    };
};
