// @injectivelabs/token-metadata/dist/tokens/tokens/cw20-tokens.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const types_1 = require("../../types");
exports.default = {
    vMOMO: {
        name: 'Momoshiki',
        logo: 'momoshiki.svg',
        symbol: 'vMOMO',
        decimals: 18,
        coinGeckoId: '',
        cw20: {
            decimals: 18,
            address: '',
            tokenType: types_1.TokenType.Cw20,
        },
    },
    vKIN: {
        name: 'Kinshiki',
        logo: 'kinshiki.svg',
        symbol: 'vKIN',
        decimals: 18,
        coinGeckoId: '',
        cw20: {
            decimals: 18,
            address: '',
            tokenType: types_1.TokenType.Cw20,
        },
    },
    vTON: {
        name: 'Toneri',
        logo: 'toneri.svg',
        symbol: 'vTON',
        decimals: 18,
        coinGeckoId: '',
        cw20: {
            decimals: 18,
            address: '',
            tokenType: types_1.TokenType.Cw20,
        },
    },
    vNAR: {
        name: 'Naruto',
        logo: 'naruto.svg',
        symbol: 'vNAR',
        decimals: 18,
        coinGeckoId: '',
        cw20: {
            decimals: 18,
            address: '',
            tokenType: types_1.TokenType.Cw20,
        },
    },
    vMAD: {
        name: 'Madara',
        logo: 'madara.svg',
        symbol: 'vMAD',
        decimals: 18,
        coinGeckoId: '',
        cw20: {
            decimals: 18,
            address: '',
            tokenType: types_1.TokenType.Cw20,
        },
    },
    vMIN: {
        name: 'Minato',
        logo: 'minato.svg',
        symbol: 'vMAD',
        decimals: 18,
        coinGeckoId: '',
        cw20: {
            decimals: 18,
            address: '',
            tokenType: types_1.TokenType.Cw20,
        },
    },
    vSPOT: {
        name: 'Spot Vault',
        logo: 'minato.svg',
        symbol: 'vSPOT',
        decimals: 18,
        coinGeckoId: '',
        cw20: {
            decimals: 18,
            address: '',
            tokenType: types_1.TokenType.Cw20,
        },
    },
    vDERI: {
        name: 'Derivative Vault',
        logo: 'minato.svg',
        symbol: 'vDERI',
        decimals: 18,
        coinGeckoId: '',
        cw20: {
            decimals: 18,
            address: '',
            tokenType: types_1.TokenType.Cw20,
        },
    },
};
