// @andromedaprotocol/andromeda.js/dist/graphql/queries/addressList.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryAddressListIncludesAddress = exports.QUERY_ADDRESS_LIST_CONTAINS_ADDRESS = void 0;
const graphql_request_1 = require("graphql-request");
const client_1 = require("../client");
exports.QUERY_ADDRESS_LIST_CONTAINS_ADDRESS = (0, graphql_request_1.gql) `
  query QUERY_ADDRESS_LIST_CONTAINS_ADDRESS(
    $contractAddress: String!
    $address: String!
  ) {
    addresslist(address: $contractAddress) {
      includesAddress(address: $address) {
        included
      }
    }
  }
`;
/**
 * Queries an address list contract whether it includes a given address
 * @param contractAddress
 * @param address
 * @returns
 */
async function queryAddressListIncludesAddress(contractAddress, address) {
    const resp = await (0, client_1.query)(exports.QUERY_ADDRESS_LIST_CONTAINS_ADDRESS, { address, contractAddress });
    return resp.addresslist.includesAddress.included;
}
exports.queryAddressListIncludesAddress = queryAddressListIncludesAddress;
