// @andromedaprotocol/andromeda.js/dist/graphql/queries/primitive.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryPrimitiveValue = exports.QUERY_PRIMITIVE_VALUE = exports.queryPrimitive = exports.QUERY_PRIMITIVE = void 0;
const graphql_request_1 = require("graphql-request");
const client_1 = require("../client");
exports.QUERY_PRIMITIVE = (0, graphql_request_1.gql) `
  query QUERY_PRIMITIVE($contractAddress: String!) {
    primitive(address: $contractAddress) {
      owner
    }
  }
`;
/**
 * Queries details about a primitive given its contract address
 * @param contractAddress The contract address of the primitive
 * @returns
 */
async function queryPrimitive(contractAddress) {
    const resp = await (0, client_1.query)(exports.QUERY_PRIMITIVE, {
        contractAddress,
    });
    return resp.primitive.owner;
}
exports.queryPrimitive = queryPrimitive;
exports.QUERY_PRIMITIVE_VALUE = (0, graphql_request_1.gql) `
  query QUERY_PRIMITIVE_VALUE($contractAddress: String!, $key: String!) {
    primitive(address: $contractAddress) {
      getValue(key: $key) {
        key
        value
      }
    }
  }
`;
/**
 * Queries the value of a given key from a given primitie contract
 * @param contractAddress The contract address of the primitive
 * @param key The key to query
 * @returns
 */
async function queryPrimitiveValue(contractAddress, key) {
    const resp = await (0, client_1.query)(exports.QUERY_PRIMITIVE_VALUE, {
        contractAddress,
        key,
    });
    return resp.primitive.getValue;
}
exports.queryPrimitiveValue = queryPrimitiveValue;
