// @andromedaprotocol/andromeda.js/dist/graphql/queries/vault.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryStrategyAddress = exports.QUERY_VAULT_STRATEGY_ADDRESS = exports.queryBalance = exports.QUERY_VAULT_BALANCE = void 0;
const client_1 = require("../client");
const graphql_request_1 = require("graphql-request");
exports.QUERY_VAULT_BALANCE = (0, graphql_request_1.gql) `
  query QUERY_VAULT_BALANCE($contractAddress: String!, $address: String!) {
    vault(contractAddress: $contractAddress) {
      balance(address: $address) {
        amount
        denom
      }
    }
  }
`;
async function queryBalance(contractAddress, address) {
    const resp = await (0, client_1.query)(exports.QUERY_VAULT_BALANCE, { contractAddress, address });
    return resp.balance;
}
exports.queryBalance = queryBalance;
exports.QUERY_VAULT_STRATEGY_ADDRESS = (0, graphql_request_1.gql) `
  query QUERY_VAULT_STRATEGY_ADDRESS(
    $contractAddress: String!
    $strategy: String!
  ) {
    vault(contractAddress: $contractAddress) {
      strategyAddress(strategy: $strategy) {
        address
        strategyType
      }
    }
  }
`;
async function queryStrategyAddress(contractAddress, strategy) {
    const resp = await (0, client_1.query)(exports.QUERY_VAULT_STRATEGY_ADDRESS, { contractAddress, strategy });
    return resp.strategyAddress;
}
exports.queryStrategyAddress = queryStrategyAddress;
