// @injectivelabs/sdk-ts/dist/core/modules/bank/msgs/MsgSend.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const coin_pb_1 = require("@injectivelabs/chain-api/cosmos/base/v1beta1/coin_pb");
const tx_pb_1 = require("@injectivelabs/chain-api/cosmos/bank/v1beta1/tx_pb");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
const MsgBase_1 = require("../../MsgBase");
/**
 * @category Messages
 */
class MsgSend extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgSend(params);
    }
    toProto() {
        const { params } = this;
        const amountToSend = new coin_pb_1.Coin();
        amountToSend.setAmount(params.amount.amount);
        amountToSend.setDenom(params.amount.denom);
        const message = new tx_pb_1.MsgSend();
        message.setFromAddress(params.srcInjectiveAddress);
        message.setToAddress(params.dstInjectiveAddress);
        message.setAmountList([amountToSend]);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmos.bank.v1beta1.MsgSend' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign(Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject())), { amount: proto
                .getAmountList()
                .map((amount) => (0, snakecase_keys_1.default)(amount.toObject())) });
        // @ts-ignore
        delete message.amount_list;
        return {
            type: 'cosmos-sdk/MsgSend',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmos.bank.v1beta1.MsgSend' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmos.bank.v1beta1.MsgSend',
            message: proto,
        };
    }
}
exports.default = MsgSend;
