// @injectivelabs/sdk-ts/dist/client/chain/types/gov.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcVote = exports.GrpcTallyResult = exports.GrpcGovernanceDepositParams = exports.GrpcGovernanceVotingParams = exports.GrpcGovernanceTallyParams = exports.GrpcProposalDeposit = exports.GrpcProposal = exports.VoteOption = exports.ProposalStatus = void 0;
const gov_pb_1 = require("@injectivelabs/chain-api/cosmos/gov/v1beta1/gov_pb");
Object.defineProperty(exports, "GrpcProposal", { enumerable: true, get: function () { return gov_pb_1.Proposal; } });
Object.defineProperty(exports, "GrpcProposalDeposit", { enumerable: true, get: function () { return gov_pb_1.Deposit; } });
Object.defineProperty(exports, "GrpcGovernanceTallyParams", { enumerable: true, get: function () { return gov_pb_1.TallyParams; } });
Object.defineProperty(exports, "GrpcGovernanceVotingParams", { enumerable: true, get: function () { return gov_pb_1.VotingParams; } });
Object.defineProperty(exports, "GrpcGovernanceDepositParams", { enumerable: true, get: function () { return gov_pb_1.DepositParams; } });
Object.defineProperty(exports, "GrpcTallyResult", { enumerable: true, get: function () { return gov_pb_1.TallyResult; } });
Object.defineProperty(exports, "GrpcVote", { enumerable: true, get: function () { return gov_pb_1.Vote; } });
var ProposalStatus;
(function (ProposalStatus) {
    ProposalStatus[ProposalStatus["PROPOSAL_STATUS_UNSPECIFIED"] = 0] = "PROPOSAL_STATUS_UNSPECIFIED";
    ProposalStatus[ProposalStatus["PROPOSAL_STATUS_DEPOSIT_PERIOD"] = 1] = "PROPOSAL_STATUS_DEPOSIT_PERIOD";
    ProposalStatus[ProposalStatus["PROPOSAL_STATUS_VOTING_PERIOD"] = 2] = "PROPOSAL_STATUS_VOTING_PERIOD";
    ProposalStatus[ProposalStatus["PROPOSAL_STATUS_PASSED"] = 3] = "PROPOSAL_STATUS_PASSED";
    ProposalStatus[ProposalStatus["PROPOSAL_STATUS_REJECTED"] = 4] = "PROPOSAL_STATUS_REJECTED";
    ProposalStatus[ProposalStatus["PROPOSAL_STATUS_FAILED"] = 5] = "PROPOSAL_STATUS_FAILED";
})(ProposalStatus = exports.ProposalStatus || (exports.ProposalStatus = {}));
var VoteOption;
(function (VoteOption) {
    VoteOption[VoteOption["VOTE_OPTION_UNSPECIFIED"] = 0] = "VOTE_OPTION_UNSPECIFIED";
    VoteOption[VoteOption["VOTE_OPTION_YES"] = 1] = "VOTE_OPTION_YES";
    VoteOption[VoteOption["VOTE_OPTION_ABSTAIN"] = 2] = "VOTE_OPTION_ABSTAIN";
    VoteOption[VoteOption["VOTE_OPTION_NO"] = 3] = "VOTE_OPTION_NO";
    VoteOption[VoteOption["VOTE_OPTION_NO_WITH_VETO"] = 4] = "VOTE_OPTION_NO_WITH_VETO";
})(VoteOption = exports.VoteOption || (exports.VoteOption = {}));
