// @injectivelabs/sdk-ts/dist/client/chain/types/insurance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcInsuranceFund = exports.GrpcRedemptionSchedule = exports.GrpcInsuranceParams = exports.OracleType = void 0;
const insurance_pb_1 = require("@injectivelabs/chain-api/injective/insurance/v1beta1/insurance_pb");
Object.defineProperty(exports, "GrpcInsuranceParams", { enumerable: true, get: function () { return insurance_pb_1.Params; } });
Object.defineProperty(exports, "GrpcInsuranceFund", { enumerable: true, get: function () { return insurance_pb_1.InsuranceFund; } });
Object.defineProperty(exports, "GrpcRedemptionSchedule", { enumerable: true, get: function () { return insurance_pb_1.RedemptionSchedule; } });
var OracleType;
(function (OracleType) {
    OracleType[OracleType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    OracleType[OracleType["BAND"] = 1] = "BAND";
    OracleType[OracleType["PRICEFEED"] = 2] = "PRICEFEED";
    OracleType[OracleType["COINBASE"] = 3] = "COINBASE";
    OracleType[OracleType["CHAINLINK"] = 4] = "CHAINLINK";
    OracleType[OracleType["RAZOR"] = 5] = "RAZOR";
    OracleType[OracleType["DIA"] = 6] = "DIA";
    OracleType[OracleType["API3"] = 7] = "API3";
    OracleType[OracleType["UMA"] = 8] = "UMA";
    OracleType[OracleType["PYTH"] = 9] = "PYTH";
    OracleType[OracleType["BANDIBC"] = 10] = "BANDIBC";
    OracleType[OracleType["PROVIDER"] = 11] = "PROVIDER";
})(OracleType = exports.OracleType || (exports.OracleType = {}));
