// @injectivelabs/sdk-ts/dist/client/indexer/types/account.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcSubaccountBalanceTransfer = exports.GrpcSubaccountPortfolio = exports.GrpcSubaccountBalance = exports.GrpcSubaccountDeposit = exports.GrpcAccountPortfolio = exports.GrpcTradingReward = exports.TransferType = void 0;
const injective_accounts_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_accounts_rpc_pb");
Object.defineProperty(exports, "GrpcTradingReward", { enumerable: true, get: function () { return injective_accounts_rpc_pb_1.Reward; } });
Object.defineProperty(exports, "GrpcAccountPortfolio", { enumerable: true, get: function () { return injective_accounts_rpc_pb_1.AccountPortfolio; } });
Object.defineProperty(exports, "GrpcSubaccountDeposit", { enumerable: true, get: function () { return injective_accounts_rpc_pb_1.SubaccountDeposit; } });
Object.defineProperty(exports, "GrpcSubaccountBalance", { enumerable: true, get: function () { return injective_accounts_rpc_pb_1.SubaccountBalance; } });
Object.defineProperty(exports, "GrpcSubaccountPortfolio", { enumerable: true, get: function () { return injective_accounts_rpc_pb_1.SubaccountPortfolio; } });
Object.defineProperty(exports, "GrpcSubaccountBalanceTransfer", { enumerable: true, get: function () { return injective_accounts_rpc_pb_1.SubaccountBalanceTransfer; } });
var TransferType;
(function (TransferType) {
    TransferType["Internal"] = "internal";
    TransferType["External"] = "external";
    TransferType["Withdraw"] = "withdraw";
    TransferType["Deposit"] = "deposit";
})(TransferType = exports.TransferType || (exports.TransferType = {}));
