// @injectivelabs/sdk-ts/dist/client/indexer/types/exchange.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcPriceLevel = exports.GrpcTokenMeta = exports.IndexerOracleType = void 0;
const injective_spot_exchange_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_spot_exchange_rpc_pb");
Object.defineProperty(exports, "GrpcTokenMeta", { enumerable: true, get: function () { return injective_spot_exchange_rpc_pb_1.TokenMeta; } });
Object.defineProperty(exports, "GrpcPriceLevel", { enumerable: true, get: function () { return injective_spot_exchange_rpc_pb_1.PriceLevel; } });
var IndexerOracleType;
(function (IndexerOracleType) {
    IndexerOracleType[IndexerOracleType["UNSPECIFIED"] = 0] = "UNSPECIFIED";
    IndexerOracleType[IndexerOracleType["BAND"] = 1] = "BAND";
    IndexerOracleType[IndexerOracleType["PRICEFEED"] = 2] = "PRICEFEED";
    IndexerOracleType[IndexerOracleType["COINBASE"] = 3] = "COINBASE";
    IndexerOracleType[IndexerOracleType["CHAINLINK"] = 4] = "CHAINLINK";
    IndexerOracleType[IndexerOracleType["RAZOR"] = 5] = "RAZOR";
    IndexerOracleType[IndexerOracleType["DIA"] = 6] = "DIA";
    IndexerOracleType[IndexerOracleType["API3"] = 7] = "API3";
    IndexerOracleType[IndexerOracleType["UMA"] = 8] = "UMA";
    IndexerOracleType[IndexerOracleType["PYTH"] = 9] = "PYTH";
    IndexerOracleType[IndexerOracleType["BANDIBC"] = 10] = "BANDIBC";
    IndexerOracleType[IndexerOracleType["PROVIDER"] = 11] = "PROVIDER";
})(IndexerOracleType = exports.IndexerOracleType || (exports.IndexerOracleType = {}));
