// @injectivelabs/sdk-ts/dist/client/indexer/types/spot.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpcSpotTrade = exports.GrpcSpotOrderHistory = exports.GrpcSpotLimitOrder = exports.GrpcSpotMarketInfo = exports.SpotOrderState = exports.SpotOrderSide = void 0;
const injective_spot_exchange_rpc_pb_1 = require("@injectivelabs/indexer-api/injective_spot_exchange_rpc_pb");
Object.defineProperty(exports, "GrpcSpotMarketInfo", { enumerable: true, get: function () { return injective_spot_exchange_rpc_pb_1.SpotMarketInfo; } });
Object.defineProperty(exports, "GrpcSpotLimitOrder", { enumerable: true, get: function () { return injective_spot_exchange_rpc_pb_1.SpotLimitOrder; } });
Object.defineProperty(exports, "GrpcSpotTrade", { enumerable: true, get: function () { return injective_spot_exchange_rpc_pb_1.SpotTrade; } });
Object.defineProperty(exports, "GrpcSpotOrderHistory", { enumerable: true, get: function () { return injective_spot_exchange_rpc_pb_1.SpotOrderHistory; } });
var SpotOrderSide;
(function (SpotOrderSide) {
    SpotOrderSide["Unspecified"] = "unspecified";
    SpotOrderSide["Buy"] = "buy";
    SpotOrderSide["Sell"] = "sell";
    SpotOrderSide["StopBuy"] = "stop_buy";
    SpotOrderSide["StopSell"] = "stop_sell";
    SpotOrderSide["TakeBuy"] = "take_buy";
    SpotOrderSide["TakeSell"] = "take_sell";
    SpotOrderSide["BuyPO"] = "buy_po";
    SpotOrderSide["SellPO"] = "sell_po";
})(SpotOrderSide = exports.SpotOrderSide || (exports.SpotOrderSide = {}));
var SpotOrderState;
(function (SpotOrderState) {
    SpotOrderState["Unfilled"] = "unfilled";
    SpotOrderState["Booked"] = "booked";
    SpotOrderState["PartialFilled"] = "partial_filled";
    SpotOrderState["PartiallyFilled"] = "partially_filled";
    SpotOrderState["Filled"] = "filled";
    SpotOrderState["Canceled"] = "canceled";
    SpotOrderState["Triggered"] = "triggered";
})(SpotOrderState = exports.SpotOrderState || (exports.SpotOrderState = {}));
