// @injectivelabs/sdk-ts/dist/core/modules/exchange/msgs/MsgBatchCancelSpotOrders.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/tx_pb");
const exchange_1 = require("../../../../types/exchange");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
const MsgBase_1 = require("../../MsgBase");
/**
 * @category Messages
 */
class MsgBatchCancelSpotOrders extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgBatchCancelSpotOrders(params);
    }
    toProto() {
        const { params } = this;
        const orderDataList = params.orders.map((order) => {
            const orderData = new tx_pb_1.OrderData();
            orderData.setMarketId(order.marketId);
            orderData.setOrderHash(order.orderHash);
            orderData.setSubaccountId(order.subaccountId);
            // TODO: Send order.orderMask instead when chain handles order mask properly.
            orderData.setOrderMask(exchange_1.OrderMask.Any);
            return orderData;
        });
        const message = new tx_pb_1.MsgBatchCancelSpotOrders();
        message.setSender(params.injectiveAddress);
        message.setDataList(orderDataList.map((o) => o));
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgBatchCancelSpotOrders' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'exchange/MsgBatchCancelSpotOrders',
            value: {
                sender: message.sender,
                data: message.data_list,
            },
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgBatchCancelSpotOrders' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/injective.exchange.v1beta1.MsgBatchCancelSpotOrders',
            message: proto,
        };
    }
}
exports.default = MsgBatchCancelSpotOrders;
