// @injectivelabs/sdk-ts/dist/core/modules/exchange/msgs/MsgCancelSpotOrder.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// import { OrderMaskMap } from '@injectivelabs/chain-api/injective/exchange/v1beta1/exchange_pb'
const tx_pb_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/tx_pb");
// import { OrderMask } from '../../../types/exchange'
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgCancelSpotOrder extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgCancelSpotOrder(params);
    }
    toProto() {
        const { params } = this;
        const message = new tx_pb_1.MsgCancelSpotOrder();
        message.setSender(params.injectiveAddress);
        message.setMarketId(params.marketId);
        message.setOrderHash(params.orderHash);
        message.setSubaccountId(params.subaccountId);
        // TODO: message.setOrderMask does not exist yet, enable this once it does.
        // message.setOrderMask(
        //   params.orderMask !== undefined ? params.orderMask : OrderMask.Any,
        // )
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgCancelSpotOrder' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'exchange/MsgCancelSpotOrder',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgCancelSpotOrder' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/injective.exchange.v1beta1.MsgCancelSpotOrder',
            message: proto,
        };
    }
}
exports.default = MsgCancelSpotOrder;
