// @injectivelabs/sdk-ts/dist/core/modules/exchange/msgs/MsgIncreasePositionMargin.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/tx_pb");
const numbers_1 = require("../../../../utils/numbers");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
const addMarginToPosition = (params) => {
    const message = new tx_pb_1.MsgIncreasePositionMargin();
    message.setSender(params.injectiveAddress);
    message.setAmount(params.amount);
    message.setMarketId(params.marketId);
    message.setSourceSubaccountId(params.srcSubaccountId);
    message.setDestinationSubaccountId(params.dstSubaccountId);
    return message;
};
/**
 * @category Messages
 */
class MsgIncreasePositionMargin extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgIncreasePositionMargin(params);
    }
    toProto() {
        const { params: initialParams } = this;
        const params = Object.assign(Object.assign({}, initialParams), { amount: (0, numbers_1.amountToCosmosSdkDecAmount)(initialParams.amount).toFixed() });
        return addMarginToPosition(params);
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgIncreasePositionMargin' }, proto.toObject());
    }
    toAmino() {
        const { params } = this;
        const proto = addMarginToPosition(params);
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'exchange/MsgIncreasePositionMargin',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgIncreasePositionMargin' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/injective.exchange.v1beta1.MsgIncreasePositionMargin',
            message: proto,
        };
    }
}
exports.default = MsgIncreasePositionMargin;
