// @injectivelabs/sdk-ts/dist/core/modules/exchange/msgs/MsgInstantSpotMarketLaunch.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/tx_pb");
const numbers_1 = require("../../../../utils/numbers");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
const createMessage = (params) => {
    const message = new tx_pb_1.MsgInstantSpotMarketLaunch();
    message.setSender(params.proposer);
    message.setQuoteDenom(params.market.quoteDenom);
    message.setTicker(params.market.ticker);
    message.setBaseDenom(params.market.baseDenom);
    message.setMinPriceTickSize(params.market.minPriceTickSize);
    message.setMinQuantityTickSize(params.market.minQuantityTickSize);
    return message;
};
/**
 * @category Messages
 */
class MsgInstantSpotMarketLaunch extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgInstantSpotMarketLaunch(params);
    }
    toProto() {
        const { params: initialParams } = this;
        const params = Object.assign(Object.assign({}, initialParams), { market: Object.assign(Object.assign({}, initialParams.market), { minPriceTickSize: (0, numbers_1.amountToCosmosSdkDecAmount)(initialParams.market.minPriceTickSize).toFixed(), minQuantityTickSize: (0, numbers_1.amountToCosmosSdkDecAmount)(initialParams.market.minQuantityTickSize).toFixed() }) });
        return createMessage(params);
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgInstantSpotMarketLaunch' }, proto.toObject());
    }
    toAmino() {
        const { params } = this;
        const proto = createMessage(params);
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'exchange/MsgInstantSpotMarketLaunch',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgInstantSpotMarketLaunch' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/injective.exchange.v1beta1.MsgInstantSpotMarketLaunch',
            message: proto,
        };
    }
}
exports.default = MsgInstantSpotMarketLaunch;
