// @injectivelabs/sdk-ts/dist/core/modules/exchange/msgs/MsgMultiExecute.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/tx_pb");
const MsgBase_1 = require("../../MsgBase");
const any_pb_1 = require("google-protobuf/google/protobuf/any_pb");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgMultiExecute extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgMultiExecute(params);
    }
    toProto() {
        const { params } = this;
        const messagesAny = params.msgs.map((m) => {
            const messageAny = new any_pb_1.Any();
            messageAny.setValue(m.toProto().serializeBinary());
            messageAny.setTypeUrl(m.toDirectSign().type);
            return messageAny;
        });
        const message = new tx_pb_1.MsgMultiExecute();
        message.setSender(params.injectiveAddress);
        message.setMsgsList(messagesAny);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgMultiExecute' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'exchange/MsgMultiExecute',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgMultiExecute' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/injective.exchange.v1beta1.MsgMultiExecute',
            message: proto,
        };
    }
}
exports.default = MsgMultiExecute;
