// @injectivelabs/sdk-ts/dist/core/modules/exchange/msgs/MsgWithdraw.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/injective/exchange/v1beta1/tx_pb");
const coin_pb_1 = require("@injectivelabs/chain-api/cosmos/base/v1beta1/coin_pb");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgWithdraw extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgWithdraw(params);
    }
    toProto() {
        const { params } = this;
        const amountCoin = new coin_pb_1.Coin();
        amountCoin.setAmount(params.amount.amount);
        amountCoin.setDenom(params.amount.denom);
        const message = new tx_pb_1.MsgWithdraw();
        message.setSender(params.injectiveAddress);
        message.setSubaccountId(params.subaccountId);
        message.setAmount(amountCoin);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgWithdraw' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'exchange/MsgWithdraw',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/injective.exchange.v1beta1.MsgWithdraw' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/injective.exchange.v1beta1.MsgWithdraw',
            message: proto,
        };
    }
}
exports.default = MsgWithdraw;
