// @injectivelabs/ts-types/dist/esm/trade.js
export var TradeExecutionType;
(function (TradeExecutionType) {
    TradeExecutionType["Market"] = "market";
    TradeExecutionType["LimitFill"] = "limitFill";
    TradeExecutionType["LimitMatchRestingOrder"] = "limitMatchRestingOrder";
    TradeExecutionType["LimitMatchNewOrder"] = "limitMatchNewOrder";
})(TradeExecutionType || (TradeExecutionType = {}));
export var TradeExecutionSide;
(function (TradeExecutionSide) {
    TradeExecutionSide["Maker"] = "maker";
    TradeExecutionSide["Taker"] = "taker";
})(TradeExecutionSide || (TradeExecutionSide = {}));
export var TradeDirection;
(function (TradeDirection) {
    TradeDirection["Buy"] = "buy";
    TradeDirection["Sell"] = "sell";
    TradeDirection["Long"] = "long";
    TradeDirection["Short"] = "short";
})(TradeDirection || (TradeDirection = {}));
export var OrderState;
(function (OrderState) {
    OrderState["Booked"] = "booked";
    OrderState["PartiallyFilled"] = "partially_filled";
    OrderState["Filled"] = "filled";
    OrderState["Canceled"] = "canceled";
    OrderState["Triggered"] = "triggered";
})(OrderState || (OrderState = {}));
