// @injectivelabs/indexer-api/injective_portfolio_rpc_pb_service.js
// package: injective_portfolio_rpc
// file: injective_portfolio_rpc.proto

var injective_portfolio_rpc_pb = require("./injective_portfolio_rpc_pb");
var grpc = require("@improbable-eng/grpc-web").grpc;

var InjectivePortfolioRPC = (function () {
  function InjectivePortfolioRPC() {}
  InjectivePortfolioRPC.serviceName = "injective_portfolio_rpc.InjectivePortfolioRPC";
  return InjectivePortfolioRPC;
}());

InjectivePortfolioRPC.AccountPortfolio = {
  methodName: "AccountPortfolio",
  service: InjectivePortfolioRPC,
  requestStream: false,
  responseStream: false,
  requestType: injective_portfolio_rpc_pb.AccountPortfolioRequest,
  responseType: injective_portfolio_rpc_pb.AccountPortfolioResponse
};

InjectivePortfolioRPC.StreamAccountPortfolio = {
  methodName: "StreamAccountPortfolio",
  service: InjectivePortfolioRPC,
  requestStream: false,
  responseStream: true,
  requestType: injective_portfolio_rpc_pb.StreamAccountPortfolioRequest,
  responseType: injective_portfolio_rpc_pb.StreamAccountPortfolioResponse
};

exports.InjectivePortfolioRPC = InjectivePortfolioRPC;

function InjectivePortfolioRPCClient(serviceHost, options) {
  this.serviceHost = serviceHost;
  this.options = options || {};
}

InjectivePortfolioRPCClient.prototype.accountPortfolio = function accountPortfolio(requestMessage, metadata, callback) {
  if (arguments.length === 2) {
    callback = arguments[1];
  }
  var client = grpc.unary(InjectivePortfolioRPC.AccountPortfolio, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onEnd: function (response) {
      if (callback) {
        if (response.status !== grpc.Code.OK) {
          var err = new Error(response.statusMessage);
          err.code = response.status;
          err.metadata = response.trailers;
          callback(err, null);
        } else {
          callback(null, response.message);
        }
      }
    }
  });
  return {
    cancel: function () {
      callback = null;
      client.close();
    }
  };
};

InjectivePortfolioRPCClient.prototype.streamAccountPortfolio = function streamAccountPortfolio(requestMessage, metadata) {
  var listeners = {
    data: [],
    end: [],
    status: []
  };
  var client = grpc.invoke(InjectivePortfolioRPC.StreamAccountPortfolio, {
    request: requestMessage,
    host: this.serviceHost,
    metadata: metadata,
    transport: this.options.transport,
    debug: this.options.debug,
    onMessage: function (responseMessage) {
      listeners.data.forEach(function (handler) {
        handler(responseMessage);
      });
    },
    onEnd: function (status, statusMessage, trailers) {
      listeners.status.forEach(function (handler) {
        handler({ code: status, details: statusMessage, metadata: trailers });
      });
      listeners.end.forEach(function (handler) {
        handler({ code: status, details: statusMessage, metadata: trailers });
      });
      listeners = null;
    }
  });
  return {
    on: function (type, handler) {
      listeners[type].push(handler);
      return this;
    },
    cancel: function () {
      listeners = null;
      client.close();
    }
  };
};

exports.InjectivePortfolioRPCClient = InjectivePortfolioRPCClient;
