// @injectivelabs/sdk-ts/dist/core/modules/staking/msgs/MsgBeginRedelegate.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tx_pb_1 = require("@injectivelabs/chain-api/cosmos/staking/v1beta1/tx_pb");
const coin_pb_1 = require("@injectivelabs/chain-api/cosmos/base/v1beta1/coin_pb");
const MsgBase_1 = require("../../MsgBase");
const snakecase_keys_1 = __importDefault(require("snakecase-keys"));
/**
 * @category Messages
 */
class MsgBeginRedelegate extends MsgBase_1.MsgBase {
    static fromJSON(params) {
        return new MsgBeginRedelegate(params);
    }
    toProto() {
        const { params } = this;
        const coinAmount = new coin_pb_1.Coin();
        coinAmount.setDenom(params.amount.denom);
        coinAmount.setAmount(params.amount.amount);
        const message = new tx_pb_1.MsgBeginRedelegate();
        message.setAmount(coinAmount);
        message.setDelegatorAddress(params.injectiveAddress);
        message.setValidatorSrcAddress(params.srcValidatorAddress);
        message.setValidatorDstAddress(params.dstValidatorAddress);
        return message;
    }
    toData() {
        const proto = this.toProto();
        return Object.assign({ '@type': '/cosmos.staking.v1beta1.MsgBeginRedelegate' }, proto.toObject());
    }
    toAmino() {
        const proto = this.toProto();
        const message = Object.assign({}, (0, snakecase_keys_1.default)(proto.toObject()));
        return {
            type: 'cosmos-sdk/MsgBeginRedelegate',
            value: message,
        };
    }
    toWeb3() {
        const amino = this.toAmino();
        const { value } = amino;
        return Object.assign({ '@type': '/cosmos.staking.v1beta1.MsgBeginRedelegate' }, value);
    }
    toDirectSign() {
        const proto = this.toProto();
        return {
            type: '/cosmos.staking.v1beta1.MsgBeginRedelegate',
            message: proto,
        };
    }
}
exports.default = MsgBeginRedelegate;
